airflow/providers/openlineage/utils/__init__.py

```diff
@@ -0,0 +1,499 @@
+
+    includes = ["duration", "try_number", "pool"]
+    casts = {
+        "map_index": lambda ti: ti.map_index
+        if hasattr(ti, "map_index") and getattr(ti, "map_index") != -1
+        else None
+    }
+
+
+class TaskInfo(InfoJsonEncodable):
+    """Defines encoding BaseOperator/AbstractOperator object to JSON."""
+
+    renames = {
+        "_BaseOperator__init_kwargs": "args",
+        "_BaseOperator__from_mapped": "mapped",
+        "_downstream_task_ids": "downstream_task_ids",
+        "_upstream_task_ids": "upstream_task_ids",
+    }
+    excludes = [
+        "_BaseOperator__instantiated",
+        "_dag",
+        "_hook",
+        "_log",
+        "_outlets",
+        "_inlets",
+        "_lock_for_execution",
+        "handler",
+        "params",
+        "python_callable",
+        "retry_delay",
+    ]
+    casts = {
+        "operator_class": lambda task: f"{get_operator_class(task).__module__}.{get_operator_class(task).__name__}",  # noqa
+        "task_group": lambda task: TaskGroupInfo(task.task_group)
+        if hasattr(task, "task_group") and getattr(task.task_group, "_group_id", None)
+        else None,
+    }
+
+
+class TaskGroupInfo(InfoJsonEncodable):
+    """Defines encoding TaskGroup object to JSON."""
+
+    renames = {
+        "_group_id": "group_id",
+    }
+    includes = [
+        "downstream_group_ids",
+        "downstream_task_ids",
+        "prefix_group_id",
+        "tooltip",
+        "upstream_group_ids",
+        "upstream_task_ids",
+    ]
+
+
+def get_airflow_run_facet(
+    dag_run: DagRun,
+    dag: DAG,
+    task_instance: TaskInstance,
+    task: BaseOperator,
+    task_uuid: str,
+):
+    return {
+        "airflow": json.loads(
+            json.dumps(
+                asdict(
+                    AirflowRunFacet(
+                        dag=DagInfo(dag),
+                        dagRun=DagRunInfo(dag_run),
+                        taskInstance=TaskInstanceInfo(task_instance),
+                        task=TaskInfo(task),
+                        taskUuid=task_uuid,
+                    )
+                ),
+                default=str,
+            )
+        )
+    }
+
+
+def get_dagrun_start_end(dagrun: DagRun, dag: DAG):
+    try:
+        return dagrun.data_interval_start, dagrun.data_interval_end
+    except AttributeError:
+        # Airflow < 2.2 before adding data interval
+        pass
+    start = dagrun.execution_date
+    end = dag.following_schedule(start)
+    return start, end or start
+
+
+def import_from_string(path: str):
+    try:
+        module_path, target = path.rsplit(".", 1)
+        module = importlib.import_module(module_path)
+        return getattr(module, target)
+    except Exception as e:
+        log.warning(e)
+        raise ImportError(f"Failed to import {path}") from e
+
+
+def try_import_from_string(path: str):
+    try:
+        return import_from_string(path)
+    except ImportError:
+        return None
```

Comment:
`from contextlib import suppress` and then

```suggestion
    with suppress(ImportError):
        return import_from_string(path)
```

Reply:
Done.